src/main/java/net/dv8tion/jda/api/exceptions/ErrorHandler.java

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright 2015-2020 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.exceptions;
+
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.internal.utils.Checks;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+/**
+ * Utility class to simplify error handling with {@link RestAction RestActions} and {@link ErrorResponse ErrorResponses}.
+ *
+ * <h2>Example</h2>
+ * <pre>{@code
+ * // Send message to user and delete it 30 seconds later, handles blocked messages in context channel.
+ * public void sendMessage(TextChannel context, User user, String content) {
+ *     user.openPrivateChannel()
+ *         .flatMap(channel -> channel.sendMessage(content))
+ *         .delay(Duration.ofSeconds(30))
+ *         .flatMap(Message::delete) // delete after 30 seconds
+ *         .queue(null, new ErrorHandler()
+ *             .ignore(ErrorResponse.UNKNOWN_MESSAGE) // if delete fails that's fine
+ *             .handle(
+ *                 (e) -> context.sendMessage("Failed to send message, you block private messages!").queue(),
+ *                 ErrorResponse.CANNOT_SEND_TO_USER)); // Fallback handling for blocked messages
```

Comment:
This needs to be updated to represent the changed argument ordering of `.handle`